test/metabase/api/dataset_test.clj

```diff
@@ -47,8 +47,10 @@
        :else
        [k v]))))
 
-(defn- most-recent-query-execution []
-  (db/select-one QueryExecution {:order-by [[:started_at :desc]]}))
+(defn- most-recent-query-execution-for-query [query]
+  (db/select-one QueryExecution
+    :hash (qp-util/query-hash query)
+    {:order-by [[:started_at :desc]]}))
```

Comment:
This should fix random test failures we were seeing on occasion

Reply:
Yeah, Oracle would bite us here if our cluster were a larger RAC cluster. Oracle does not guarantee ordering of results (due to parallelism in RAC) unless an `ORDER BY` is used.